### PR TITLE
Speech command recognition - MatchboxNet

### PR DESCRIPTION
**Pull request by jovaroo**
Adds an implementation of MatchboxNet, which can be used for keyword spotting or other limited-vocabulary speech recognition tasks.

### CODE REVIEW

**Comment by passalis** [comment]
@jovaroo testing is now available again. So, when you are ready, you can use the "test sources" label to run the style checks and "test tools" to run the tool tests.

**Comment by jovaroo** [comment]
@passalis The tests have now successfully run and the PR is ready for final review!

**Review by passalis — DISMISSED** [review]
Thanks!

File: tests/sources/tools/perception/speech_recognition/matchboxnet/test_matchboxnet.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+    def __getitem__(self, item):
+        return np.ones(TEST_SIGNAL_LENGTH), np.random.choice(TEST_CLASSES_N)
+
+
+class MatchboxNetTest(unittest.TestCase):
+    learner = None
+
+    @classmethod
+    def setUpClass(cls):
```
Comment (stefaniapedrazzi) [inline tests/sources/tools/perception/speech_recognition/matchboxnet/test_matchboxnet.py:56]: Could you please add a log message specifying which test is running?
Something like this:
```suggestion
    def setUpClass(cls):
        print("\n\n**********************************\nTEST Speech command recognition MatchboxNetLearner\n"
              "**********************************")
```

**Review by stefaniapedrazzi — CHANGES_REQUESTED** [review]
Thank you!
Everything is fine for me as well.
It would just be good to log the test name at the very beginning.

**Comment by jovaroo** [comment]
@stefaniapedrazzi Applied the suggested change!

**Review by stefaniapedrazzi — APPROVED** [review]
Thank you!

**Review by passalis — APPROVED** [review]

**Pull request merged**